emcc.py

```diff
@@ -717,6 +717,18 @@ def uniquename(name):
   # specified_target is the user-specified one, target is what we will generate
   if specified_target:
     target = specified_target
+    # show a clear error if the user sets a specified target to something
+    # no-existing like NONEXISTENT_DIR/something. Python will give an error
+    # "no such file" when we try to write to that file, which can be
+    # confusing. worse, this will happen on the first file we try to write
+    # to, which may be one of the helper files (.wasm, .mem, etc.), which
+    # is even more confusing.
+    # (note that we must check that dirname is not the empty string - if the
+    # target is just a basename, then dirname is '', which os.path.exists would
+    # report as not existing)
+    dirname = os.path.dirname(target)
```

Comment:
I don't think the final sentence in brackets is necessary.  It obvious enough (I think) what `if dirname and not os.path.exists(dirname):` is doing.

Reply:
How about the following comment
```
Check for the existence of the output directory now, to avoid having to do so repeatedly when each of the various output files (.mem, .wasm, etc) is written.   This gives a more useful error message than the IOError and python backtrace that users would otherwise see.
```